extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic.authorization;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.druid.java.util.common.logger.Logger;
+import io.druid.security.basic.db.BasicSecurityStorageConnector;
+import io.druid.server.security.Access;
+import io.druid.server.security.Action;
+import io.druid.server.security.AuthenticationResult;
+import io.druid.server.security.Authorizer;
+import io.druid.server.security.Resource;
+import io.druid.server.security.ResourceAction;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@JsonTypeName("basic")
+public class BasicRoleBasedAuthorizer implements Authorizer
+{
+  private static final Logger log = new Logger(BasicRoleBasedAuthorizer.class);
+
+  private final BasicSecurityStorageConnector dbConnector;
+
+  @JsonProperty
+  private final boolean remapAuthNames;
+
+  @JsonCreator
+  public BasicRoleBasedAuthorizer(
+      @JacksonInject BasicSecurityStorageConnector dbConnector,
+      @JsonProperty("remapAuthNames") Boolean remapAuthNames
+  )
+  {
+    this.dbConnector = dbConnector;
+    this.remapAuthNames = remapAuthNames == null ? false : remapAuthNames;
+  }
+
+  @Override
+  public Access authorize(
+      AuthenticationResult authenticationResult, Resource resource, Action action
+  )
+  {
+    if (authenticationResult == null) {
+      return new Access(false);
```

Comment:
Any error message like 'null authenticationResult'?

Reply:
I changed this to throw an IAE, the authenticationResult should never be null here and indicates a bug if it is